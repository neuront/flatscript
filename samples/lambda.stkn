console.log(((): 42)())
console.log(((a): a)(42))

make_pair: (x, y):
    return (s):
        return s(x, y)

first: (x, y): x

m: 10
x: make_pair(m, 6)
console.log(x(first))
console.log(x((x, y): y))

y: make_pair(false, true)
console.log(y(first))
console.log(y((x, y): y))
