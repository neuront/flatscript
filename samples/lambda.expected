42
42
10
6
false
true
